Tests/UnitTests/MessageBirdUnitTests/Resources/ClientVerifyTest.cs

```diff
@@ -0,0 +1,151 @@
+﻿using MessageBird.Net;
+using MessageBird.Objects;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using WireMock.Server;
+
+namespace MessageBirdTests.Resources
+{
+    /**
+     * Using this class to be able to change the base url of the request.
+     */ 
+    internal class VerifyWithUrl : MessageBird.Resources.Resource
+    {
+        public VerifyWithUrl(Verify verify)
+            : base("verify", verify)
+        {
+        }
+
+        private string VerifyBaseUrl;
+
+        private string Token
+        {
+            get
+            {
+                return (Object != null) ? ((Verify)Object).Token : null;
+            }
+        }
+
+        private bool HasToken
+        {
+            get
+            {
+                return !string.IsNullOrEmpty(Token);
+            }
+        }
```

Comment:
I don't think if you need this property - you could just check if the Token is null or not on L40 where you are using this